src/Functions/FunctionBase64Conversion.cpp

```diff
@@ -0,0 +1,93 @@
+#include "config.h"
+#if USE_BASE64
```

Comment:
Thinking about it, we could keep it simple and move the implementation of both functions in this file into the header (FunctionBase64Conversion.h). Headers in src/Functions/ are usually only included from at most a handful of source files (base64Decode.cpp, base64Encode.cpp, tryBase64Decode.cpp) and not from other headers (and/or possibly transitively from more headers). That will also make the forward declarations of both functions obsolete (l. 31/32 of FunctionBase64Conversion.h).

Reply:
> Thinking about it, we could keep it simple and move the implementation of both functions in this file into the header (FunctionBase64Conversion.h). Headers in src/Functions/ are usually only included from at most a handful of source files (base64Decode.cpp, base64Encode.cpp, tryBase64Decode.cpp) and not from other headers (and/or possibly transitively from more headers). That will also make the forward declarations of both functions obsolete (l. 31/32 of FunctionBase64Conversion.h).

In that case linker errors on duplicate symbols:
[build] ld.lld-18: error: duplicate symbol: DB::preprocessBase64Url(std::__1::span<char8_t const, 18446744073709551615ul>)
[build] >>> defined at FunctionBase64Conversion.h:32 (/home/mgorshkov/git/ADQM-1893/src/Functions/FunctionBase64Conversion.h:32)
[build] >>>            src/Functions/CMakeFiles/clickhouse_functions_obj.dir/./base64Decode.cpp.o:(DB::preprocessBase64Url(std::__1::span<char8_t const, 18446744073709551615ul>))
[build] >>> defined at FunctionBase64Conversion.h:32 (/home/mgorshkov/git/ADQM-1893/src/Functions/FunctionBase64Conversion.h:32)
[build] >>>            src/Functions/CMakeFiles/clickhouse_functions_obj.dir/./base64Encode.cpp.o:(.text+0x0)
Let's make the functions inline, if it's ok (probably, code bloat is a side effect).

Reply:
That's fine, yes.